src/main/java/org/truffleruby/language/yield/YieldExpressionNode.java

```diff
@@ -97,4 +107,18 @@ private YieldNode getYieldNode() {
         return yieldNode;
     }
 
+    private void warnIfYieldInSingletonClass(boolean isSingletonClass) {
+        if (isSingletonClass) {
```

Comment:
Let's move this `if` in the caller, so this method only contains the logic to warn.